pkg/controllers/metrics/podmetrics/podmetrics.go

```diff
@@ -0,0 +1,185 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podmetrics
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"knative.dev/pkg/logging"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	podname             = "name"
+	podnamespace        = "namespace"
+	ownerselflink       = "owner"
+	podhostname         = "node"
+	podprovisioner      = "provisioner"
+	podhostzone         = "zone"
+	podhostarchitecture = "arch"
+	podhostcapacitype   = "capacitytype"
+	podhostinstancetype = "instancetype"
+	podphase            = "phase"
```

Comment:
Can we make these CamelCaseVariableNames and more descriptive of how they're used?
e.g.
`PodNameLabelKey`
`PodNamespaceLabelKey`

Reply:
good point. should have caught this before.

Reply:
done